src/pca.ts

```diff
@@ -12,9 +21,29 @@ import { Matrix, MatrixTransposeView, EVD, SVD, NIPALS } from 'ml-matrix';
  * @param {boolean} [options.ignoreZeroVariance=false] - ignore columns with zero variance if `scale` is `true`.
  * */
 export class PCA {
-  constructor(dataset, options = {}) {
-    if (dataset === true) {
-      const model = options;
+  private center: boolean;
+  private scale: boolean;
+  private excludedFeatures: number[];
+  /* eslint-disable @typescript-eslint/naming-convention */
+  private U: Matrix | null = null;
+  private S: number[] | null = null;
+  private R: any;
+  private means: number[] | null;
+  private stdevs: number[] | null;
+
+  public constructor(
+    dataset?: MaybeMatrix,
+    options: {
+      isCovarianceMatrix?: boolean;
+      method?: string;
+      nCompNIPALS?: number;
+      center?: boolean;
+      scale?: boolean;
+      ignoreZeroVariance?: boolean;
+    } = {},
```

Comment:
For objects, we like to define the types in interfaces with names. See https://github.com/mljs/pca/blob/master/pca.d.ts (ignore the `I` at the beginning of the interface names. It's a convention that we do not use anymore.)

Reply:
For example here, it should be:

```
export interface PCAOptions {
  isCovarianceMatrix?: boolean;
  method?: 'SVD' | 'NIPALS' | 'covarianceMatrix';
  center?: boolean;
  scale?: boolean;
  nCompNIPALS?: number;
  ignoreZeroVariance?: boolean;
}
```

```
options: PCAOptions = {},
```